packages/cli/test/sim/backup_eth_provider.test.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {join} from "node:path";
+import {activePreset} from "@lodestar/params";
+import {nodeAssertion} from "../utils/simulation/assertions/nodeAssertion.js";
+import {CLIQUE_SEALING_PERIOD, SIM_TESTS_SECONDS_PER_SLOT} from "../utils/simulation/constants.js";
+import {CLClient, ELClient} from "../utils/simulation/interfaces.js";
+import {SimulationEnvironment} from "../utils/simulation/SimulationEnvironment.js";
+import {getEstimatedTimeInSecForRun, getEstimatedTTD, logFilesDir} from "../utils/simulation/utils/index.js";
+import {connectAllNodes, waitForSlot} from "../utils/simulation/utils/network.js";
+
+const genesisSlotsDelay = 20;
+const altairForkEpoch = 2;
+const bellatrixForkEpoch = 4;
+// Make sure bellatrix started before TTD reach
+const additionalSlotsForTTD = activePreset.SLOTS_PER_EPOCH - 2;
+const runTillEpoch = 6;
+const syncWaitEpoch = 2;
+
+const runTimeoutMs =
+  getEstimatedTimeInSecForRun({
+    genesisSlotDelay: genesisSlotsDelay,
+    secondsPerSlot: SIM_TESTS_SECONDS_PER_SLOT,
+    runTill: runTillEpoch + syncWaitEpoch,
+    // After adding Nethermind its took longer to complete
+    graceExtraTimeFraction: 0.3,
+  }) * 1000;
+
+const ttd = getEstimatedTTD({
+  genesisDelay: genesisSlotsDelay,
+  bellatrixForkEpoch: bellatrixForkEpoch,
+  secondsPerSlot: SIM_TESTS_SECONDS_PER_SLOT,
+  cliqueSealingPeriod: CLIQUE_SEALING_PERIOD,
+  additionalSlots: additionalSlotsForTTD,
+});
+
+const env = SimulationEnvironment.initWithDefaults(
+  {
+    id: "backup-eth-provider",
+    logsDir: join(logFilesDir, "backup-eth-provider"),
+    chainConfig: {
+      ALTAIR_FORK_EPOCH: altairForkEpoch,
+      BELLATRIX_FORK_EPOCH: bellatrixForkEpoch,
+      GENESIS_DELAY: genesisSlotsDelay,
+      TERMINAL_TOTAL_DIFFICULTY: ttd,
+    },
+  },
+  [{id: "node-1", cl: CLClient.Lodestar, el: ELClient.Geth, keysCount: 32, mining: true}]
+);
+
+env.tracker.register({
+  ...nodeAssertion,
+  match: ({slot}) => {
+    return slot === 1 ? {match: true, remove: true} : false;
+  },
+});
+
+const newNode1 = env.createNodePair({
+  id: "node-2",
+  cl: {type: CLClient.Lodestar, options: {engineUrls: [env.nodes[0].el.engineRpcUrl]}},
+  el: ELClient.Geth,
+  keysCount: 0,
+});
+
+const newNode2 = env.createNodePair({
+  id: "node-3",
+  cl: {type: CLClient.Lodestar, options: {engineUrls: [env.nodes[0].el.engineRpcUrl]}},
+  el: ELClient.Geth,
+  keysCount: 0,
+});
+
+env.nodes.push(newNode1);
+env.nodes.push(newNode2);
+
+await env.start({runTimeoutMs});
+await connectAllNodes(env.nodes);
+
+await waitForSlot(env.clock.getLastSlotOfEpoch(1), env.nodes, {silent: true, env});
+
+await newNode1.el.job.stop();
+await newNode2.el.job.stop();
```

Comment:
What is this test testing exactly? A backup test should
- start CL 1 pointing to EL [1,2]
- ensure CL 1 progress ok targeting EL 1
- stop EL 1
- ensure CL 1 progresses ok targeting EL 2

Reply:
This test is doing exactly the same. 

1. start CL/EL (1)
2. start CL/EL (2) with additional EL url pointing to EL1
3. start CL/EL (3) with additional EL url pointing to EL1
4. Wait for 1 epoch and assert nodes are progressing fine. 
5. Stop EL2
6. Stop EL3
7. Test that CL2 and CL3 are progressing fine and have same head as CL1. 

Do you think this approach needs to be changed?

Reply:
I update some code and added the comments. Also tried the nodes without additional engine URLs to make sure the test fails.

Reply:
I made some changes and added the comments as well. Please re-review.